HACKING.md

```diff
@@ -0,0 +1,118 @@
+
+LU used two different types of servers to communicate. These are: 
+* The authentication server (Uchu.Auth), which handles login and redirection to the world server. 
+* The world server, which handles two things
+  * Character selection (Uchu.Char)
+  * Main game (Uchu.World)
+
+**Note:** A world server in Uchu only handles one world but there can be multiple world servers in one Uchu instance. They can even serve the same map if one world server has reached the maximum amount of player. 
+
+The address of the auth server is configured in the `boot.cfg` file in the installation directory of the client. All other servers can have any address because the previous server sends the address to the client.
+
+There were also two other server types that aren't implemented in Uchu. These are
+* The Chat server which was never used because the world sever handles chat messages
+* The UGC (user generated content) server. This handles stuff like custom icons for modular rockets and cars. The UGC Server address is also set in `boot.cfg`
+
+## The repo structure
+As you probably already noticed the Uchu repository is quite complex. Therefore this paragraph will explain what the different folders contain and what they are for.
+
+There are 3 submodules (marked with a blue text color in GitHub) in the repo.
+* [InfectedRose](https://github.com/Wincent01/InfectedRose) - A library to work with LEGO® Universe files
+* [NexusLogging](https://github.com/TheNexusAvenger/Nexus-Logging) - A logging library for .NET
+* [RakDotNet](https://github.com/UchuServer/RakDotNet) - A library for the RakNet protocol that LU uses
+
+Then there are also the folders that contain the actual Uchu code. These are:
+
+* **Uchu.Api** - It contains code for an api with which the Uchu instance can be controlled. There is a [Cli Tool](https://github.com/UchuServer/cli) that uses this api
+* **Uchu.Auth** - This contains the auth server
+* **Uchu.Char** - This contains the char server
+* **Uchu.Core.Test** - Unit Tests for Uchu.Core
+* **Uchu.Core** - Contains stuff that is used all over the project, like logging, io, database access, etc
+* **Uchu.Instance** - The base for all servers that Uchu runs (like auth, char, world)
+* **Uchu.Master** - This contains the main process that starts and controlls all the servers
+* **Uchu.Navigation** - Code for pathfinding (used by the world server)
+* **Uchu.Physics.Test**- Unit Tests for Uchu.Physics
+* **Uchu.Physics** - Code for all kinds of physics calculations (used by the world server)
+* **Uchu.Python** - Python interpreter for scripts
+* **Uchu.Sso** - Code for lcdr's [Single-Sign-On](https://github.com/lcdr/sso_auth)
+* **Uchu.StandardScripts** - C# versions of server scripts (most of the original lua server scripts are not available)
```

Comment:
I'd reword this to say "C# re-implementations of lua server scripts (most of which are not available in the client)"